integration_tests/src/tests/multisig_test.rs

```diff
@@ -56,9 +56,7 @@ pub fn proposal_hash_test(v: &dyn VM) {
         approved: vec![alice],
         params: RawBytes::default(),
     };
-    let wrong_hash =
-        compute_proposal_hash(&wrong_tx, &MockRuntime::new(&DynBlockstore::wrap(v.blockstore())))
-            .unwrap();
+    let wrong_hash = compute_proposal_hash(&wrong_tx, &MockRuntime::new()).unwrap();
```

Comment:
This one looks like it was passing through a specific, probably non-empty store. We need to preserve that behaviour. Add a `new_with_store` constructor for the Mock Runtime so that one can be provided externally if desired.

Reply:
This is really tricky to do. The `blockstore ` method on the VM trait returns an `fvm_ipld_blockstore` but the `Mock Runtime` needs a `Rc<memory blockstore>` to work with.

I looked into changing the VM trait to get the `blockstore` method to return a concrete `MemoryBlockstore` but we can't get the `vm_api` create to depend on the `runtime` crate as this creates a cyclical dependency.

Should we move the "test blockstores" to the "vm_api" package -> get the VM trait to return a concrete Memory Blockstore and take it from there or do you have an alternate suggestion in mind ?

Reply:
I don't want to move the blockstore into the vm_api crate.

Can we get the VM to hold a reference to an `impl Blockstore` instead of the concrete type?

Reply:
Wait this whole thing is silly. This is an integration test constructing a unit test runtime for the sole purpose of getting at a `hash_blake2b` function.  Just rewrite this test not to use MockRuntime.

Reply:
The TestVM has Primitives too - use that

Reply:
This is done.